File: src/main/java/org/nordix/simplepki/domain/model/SerialNumberConverter.java
package org.nordix.simplepki.domain.model;

import lombok.experimental.UtilityClass;

import java.math.BigInteger;

@UtilityClass
public class SerialNumberConverter {

    public static final String REGEXP = "^[\\dA-F]{16}$";

    public String toString(BigInteger serialNumber) {
        return toString(serialNumber.longValue());
    }

    public String toString(long serialNumber) {
        return String.format("%016X", serialNumber);
    }

    public long fromString(String serialNumber) {
        return Long.parseLong(serialNumber, 16);
    }
}

File: src/main/kotlin/org/nordix/simplepki/domain/model/SerialNumberConverter.kt
package org.nordix.simplepki.domain.model

import java.math.BigInteger
import java.util.*

object SerialNumberConverter {

    @JvmStatic
    fun toString(serialNumber: BigInteger): String {
        return toString(serialNumber.toLong())
    }

    @JvmStatic
    fun toString(serialNumber: Long): String {
        return String.format(Locale.getDefault(Locale.Category.FORMAT), "%016X", serialNumber)
    }

    fun fromString(serialNumber: String): Long {
        return serialNumber.toLong(HEX_RADIX)
    }

    const val REGEXP = "^[\\dA-F]{16}$"
    private const val HEX_RADIX = 16
}
